tensorflow/contrib/losses/python/losses/loss_ops.py

```diff
@@ -645,8 +647,12 @@ def cosine_distance(
     ValueError: If `predictions` shape doesn't match `labels` shape, or
       `weights` is `None`.
   """
-  if dim is None:
-    raise ValueError("`dim` cannot be None.")
+  if dim is not None:
+    if axis is not None:
+      raise ValueError("Cannot specify both 'axis' and 'dim'")
+    axis = dim
+  if axis is None and dim is None:
+    raise ValueError("`axis` and `dim` cannot both be None.")
```

Comment:
Can you change the error message here to "You must specify 'axis'."

Since dim is deprecated, no need to mention it.

Reply:
Thanks. Done.